File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Audio.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Audio implements ITableOperate {
 public final static String TAB_FILE_DATA = "/audio.txt";
 public static final int ID_NAME = 0 ;
public static final int ID_SUFFIX = 1 ;
public static final int ID_PLAYTIMES = 2 ;
public static final int ID_DELAYTIME = 3 ;
public static final int ID_TYPE = 4 ;
public static final int MAX_RECORD = 5 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private float m_Delaytime;
 public float GetDelaytime() { return m_Delaytime;}

private String m_Name;
 public String GetName() { return m_Name;}

private int m_Playtimes;
 public int GetPlaytimes() { return m_Playtimes;}

private String m_Suffix;
 public String GetSuffix() { return m_Suffix;}

private int m_Type;
 public int GetType() { return m_Type;}

public boolean LoadTable(HashMap<Integer, Table_Audio> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Audio> _hash = (HashMap<Integer, Table_Audio> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Audio _values = new Table_Audio();
 _values.m_Delaytime =  Float.parseFloat((String)valuesList.get(ID_DELAYTIME));
_values.m_Name =  (String)valuesList.get(ID_NAME);
_values.m_Playtimes =  Integer.parseInt((String)valuesList.get(ID_PLAYTIMES));
_values.m_Suffix =  (String)valuesList.get(ID_SUFFIX);
_values.m_Type =  Integer.parseInt((String)valuesList.get(ID_TYPE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/tabletools/CodeTable/CSharp/Table_Audio.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Audio : ITableOperate
{ private const string TAB_FILE_DATA = "audio.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_NAME,
ID_SUFFIX,
ID_PLAYTIMES,
ID_DELAYTIME,
ID_TYPE,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private float m_Delaytime;
 public float Delaytime { get{ return m_Delaytime;}}

private string m_Name;
 public string Name { get{ return m_Name;}}

private int m_Playtimes;
 public int Playtimes { get{ return m_Playtimes;}}

private string m_Suffix;
 public string Suffix { get{ return m_Suffix;}}

private int m_Type;
 public int Type { get{ return m_Type;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Audio _values = new Tab_Audio();
 _values.m_Delaytime =  Convert.ToSingle(valuesList[(int)_ID.ID_DELAYTIME] as string);
_values.m_Name =  valuesList[(int)_ID.ID_NAME] as string;
_values.m_Playtimes =  Convert.ToInt32(valuesList[(int)_ID.ID_PLAYTIMES] as string);
_values.m_Suffix =  valuesList[(int)_ID.ID_SUFFIX] as string;
_values.m_Type =  Convert.ToInt32(valuesList[(int)_ID.ID_TYPE] as string);

 _hash.Add(nKey,_values); }


}
}

